### PR TITLE
[libdatachannel] update to 0.17.1

### LINKED ISSUE #24266: [libdatachannel] update to 0.17.1
Library name: libdatachannel

New version number: 0.17.1

https://github.com/paullouisageneau/libdatachannel/releases

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fix #24266

Update libdatachannel to the latest version 0.17.1

All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

Depends on pr #24288

### CODE REVIEW

File: ports/libdatachannel/portfile.cmake [hunk 1]
```diff
@@ -28,7 +28,7 @@ vcpkg_cmake_configure(
         ${FEATURE_OPTIONS}
         -DUSE_SYSTEM_SRTP=ON
         -DNO_EXAMPLES=ON
-        -DNO_TESTS=ON
+        -DNO_TESTS=OFF
```
Comment (Thomas1664) [inline ports/libdatachannel/portfile.cmake:31]:
```suggestion
        -DNO_TESTS=ON
```
Why enable tests?

Comment (JonLiu1993) [reply ports/libdatachannel/portfile.cmake:31]: @Thomas1664 , My fault , I read it wrong, I thought it was to disable test. Thank you very munch!

**Comment by Nemirtingas** [comment]
You might want to simplify the patch here:
https://github.com/microsoft/vcpkg/blob/master/ports/libdatachannel/0001-fix-for-vcpkg.patch#L65-L66

By only enabling USE_SYSTEM_JUICE in the vcpkg options:
https://github.com/paullouisageneau/libdatachannel/blob/master/CMakeLists.txt#L11

https://github.com/microsoft/vcpkg/blob/master/ports/libdatachannel/portfile.cmake#L25-L32

**Comment by JonLiu1993** [comment]
> You might want to simplify the patch here: https://github.com/microsoft/vcpkg/blob/master/ports/libdatachannel/0001-fix-for-vcpkg.patch#L65-L66
> 
> By only enabling USE_SYSTEM_JUICE in the vcpkg options: https://github.com/paullouisageneau/libdatachannel/blob/master/CMakeLists.txt#L11
> 
> https://github.com/microsoft/vcpkg/blob/master/ports/libdatachannel/portfile.cmake#L25-L32

@Nemirtingas ,Thanks for your suggestion. libjuice is vcpkg's own library. When we use it, we will use vcpkg's own library first. If not, we will consider system or third-party libraries, because it will be more stable and conducive to later maintenance.

**Review by FrankXie05 — APPROVED** [review]

**Comment by Nemirtingas** [comment]
> > You might want to simplify the patch here: https://github.com/microsoft/vcpkg/blob/master/ports/libdatachannel/0001-fix-for-vcpkg.patch#L65-L66
> > By only enabling USE_SYSTEM_JUICE in the vcpkg options: https://github.com/paullouisageneau/libdatachannel/blob/master/CMakeLists.txt#L11
> > https://github.com/microsoft/vcpkg/blob/master/ports/libdatachannel/portfile.cmake#L25-L32
> 
> @Nemirtingas ,Thanks for your suggestion. libjuice is vcpkg's own library. When we use it, we will use vcpkg's own library first. If not, we will consider system or third-party libraries, because it will be more stable and conducive to later maintenance.

Isn't the code the same ? When using system libjuice, the cmake file calls find_package(LibJuice REQUIRED) and in the patch we also call find_package(LibJuice REQUIRED).

**Comment by JonLiu1993** [comment]
> > > You might want to simplify the patch here: https://github.com/microsoft/vcpkg/blob/master/ports/libdatachannel/0001-fix-for-vcpkg.patch#L65-L66
> > > By only enabling USE_SYSTEM_JUICE in the vcpkg options: https://github.com/paullouisageneau/libdatachannel/blob/master/CMakeLists.txt#L11
> > > https://github.com/microsoft/vcpkg/blob/master/ports/libdatachannel/portfile.cmake#L25-L32
> > 
> > 
> > @Nemirtingas ,Thanks for your suggestion. libjuice is vcpkg's own library. When we use it, we will use vcpkg's own library first. If not, we will consider system or third-party libraries, because it will be more stable and conducive to later maintenance.
> 
> Isn't the code the same ? When using system libjuice, the cmake file calls find_package(LibJuice REQUIRED) and in the patch we also call find_package(LibJuice REQUIRED).

@Nemirtingas , Thanks for the suggestion, I looked into it again and indeed it is possible to use the option to simplify the patch, I will submit the changes later

**Review by FrankXie05 — APPROVED** [review]

**Review by raghavthind2005 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**